scipy/stats/_entropy.py

```diff
@@ -40,7 +45,7 @@ def entropy(pk, qk=None, base=None, axis=0):
 
     Returns
     -------
-    S : float
+    S : {float, array_like}
```

Comment:
This should only be `array_like`. It includes floating points.

```suggestion
    S : array_like
```

Reply:
Even though it is correct, is it informative?
I see array_like as very informative when specifying types for input, but when reading the docs, I would expect return types to be more informative. For instance, I would want to know if I get receive an `array scalar` of and array of dim=0 as specified here:
https://numpy.org/devdocs/glossary.html#term-array-scalar

If you think this is good enough, I have know problem committing your suggestion.

Reply:
I think there is currently a trade-off between consistency and preciseness. Most of the stats functions use `array_like` as return types (and others that don't either use `float` or `ndarray`). I don't have a problem with `{float, array_like}` to be more precise. It's just that I don't see how this would help the user: it's still not clear when a `float` is returned and when `array_like` is returned unless explicitly specified in the docs. I will let you choose if you want to go for `{float, array_like}` or just `array_like`: I don't have any strong preference/opinion here.

Reply:
personally, to me it seems intuitive the the actual return type would depend on actual dimensionality, so I'll leave it as is if it is up to me.